src/libs/Localize/index.ts

```diff
@@ -100,49 +91,38 @@ function translate(desiredLanguage = CONST.LOCALES.DEFAULT, phraseKey, phrasePar
 
 /**
  * Uses the locale in this file updated by the Onyx subscriber.
- *
- * @param {String|Array} phrase
- * @param {Object} [variables]
- * @returns {String}
  */
-function translateLocal(phrase, variables) {
-    return translate(BaseLocaleListener.getPreferredLocale(), phrase, variables);
+function translateLocal<TKey extends TranslationPaths>(phrase: TKey, ...variables: PhraseParameters<Phrase<TKey>>) {
+    return translate(BaseLocaleListener.getPreferredLocale(), phrase, ...variables);
 }
 
 /**
  * Return translated string for given error.
- *
- * @param {String|Array} message
- * @returns {String}
  */
-function translateIfPhraseKey(message) {
-    if (_.isEmpty(message)) {
+function translateIfPhraseKey(message: string | [string, Record<string, unknown> & {isTranslated?: true}]): string {
+    if (!message || (Array.isArray(message) && message.length > 0)) {
         return '';
     }
 
     try {
         // check if error message has a variable parameter
-        const [phrase, variables] = _.isArray(message) ? message : [message];
+        const [phrase, variables] = Array.isArray(message) ? message : [message];
 
         // This condition checks if the error is already translated. For example, if there are multiple errors per input, we handle translation in ErrorUtils.addErrorMessage due to the inability to concatenate error keys.
-
-        if (variables && variables.isTranslated) {
+        if (variables?.isTranslated) {
             return phrase;
         }
 
-        return translateLocal(phrase, variables);
+        return translateLocal(phrase as TranslationPaths, variables as never);
```

Comment:
Why are we casting `variables as never` here?

Reply:
@arosiclair We're casting `variables` as `never` to bypass TypeScript's strict type checking. `translateIfPhraseKey` is being used dynamically, e.g., with errors that comes from the backend which we don't have static types for. 

It allows `translateIfPhraseKey` to call the strictly typed `translateLocal` function without type errors. Yep, it does reduce type safety for variables, but looking at the implementation of `translation` and `translationLocal` it shouldn't be an issue.